src/basic/path-lookup.c

```diff
@@ -715,8 +717,13 @@ int lookup_paths_init(
         if (r < 0)
                 return -ENOMEM;
 
+        r = hashmap_ensure_allocated(&link_cache, &string_hash_ops);
```

Comment:
string_hash_ops is not the ideal choice here. you have paths, hence you want to use something like `path_hash_ops_free` or `path_hash_ops`, i.e. a hash function and comparison function that understands paths

Reply:
Please use `path_hash_ops_free` in new code, and `hashmap_free` and not one of the explicit variants.